src/node-lib/curriculum-api-2023/queries/subjectListing/processLesson.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  subjects,
+  subjectSlugs,
+  SyntheticUnitvariantLessons,
+} from "@oaknational/oak-curriculum-schema";
+
+interface UnprocessedSubject {
+  [key: string]: {
+    subjectTitle: typeof subjects | string;
+    subjectSlug: typeof subjectSlugs | string;
+    programmeSlug: string;
+    unitIds: Set<number>;
+    lessonSlugs: Set<string>;
+    programmeSlugs: Set<string>;
+  };
+}
+
+interface ProcessedSubject {
+  subjectTitle: typeof subjects | string;
+  subjectSlug: typeof subjectSlugs | string;
+  programmeSlug: string;
+  unitCount: number;
+  lessonCount: number;
+  programmeCount: number;
+}
+
+export const processLessons = (
+  lessons: SyntheticUnitvariantLessons[],
+): ProcessedSubject[] => {
+  const subjects = {} as UnprocessedSubject;
+
+  for (const lesson of lessons) {
+    // eslint-disable-next-line prefer-const
+    let { programme_slug, lesson_slug } = lesson;
+    const { subject, keystage_slug, tier_slug, examboard_slug } =
+      lesson.programme_fields;
+    const { unit_id } = lesson.unit_data;
+    const newProgrammeSlug = programme_slug;
+    // If there's a tier_slug or examboard_slug, slice the programme_slug after the keystage_slug
```

Comment:
would constructing the base programme slug from `{subject}-{phase}-{ks}` from programme fields work?